src/partition.rs

```diff
@@ -143,6 +141,21 @@ pub enum BlockSize {
   BLOCK_INVALID,
 }
 
+impl PartialOrd for BlockSize {
+  fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+    use std::cmp::Ordering::{Equal, Greater, Less};
+    match (
+      self.width().cmp(&other.width()),
+      self.height().cmp(&other.height()),
+    ) {
+      (Greater, Less) | (Less, Greater) => None,
+      (Equal, Equal) => Some(Equal),
+      (Greater, _) | (_, Greater) => Some(Greater),
+      (Less, _) | (_, Less) => Some(Less),
```

Comment:
I'm thinking the `or` conditions in the last two match arms are unreachable, because they're handled by the first match arm.

Reply:
With this change, the match is incomplete:
```
patterns `(Equal, Less)` and `(Equal, Greater)` not covered
```
I distilled [an example to explore](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a37ce7e35a15fe9e4a7bfaaf4a13ed5e).

Reply:
Ohh you're right, good call.